.github/workflows/publish.yml

```diff
@@ -27,25 +27,100 @@ jobs:
             echo "Error: tag does not match semver"
             exit 1
           fi
-  publish:
+
+  # If we use the viamrobotics/build-action@v1, it farms out compilation to Linux machines no
+  # matter which OS you're targeting (which is fine for Go and Python, but not for C++).
+  # Furthermore, these Linux machines do not cache build artifacts from one run to another. So
+  # instead, we're going to build everything on Github's action runners directly.
+  build-windows:
+    name: Build for Windows
     needs: validate-tag
-    name: Build and Upload module
-    runs-on: ubuntu-latest
+    runs-on: windows-2019
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+      - name: Restore cached libraries
+        uses: actions/cache@v3
+        with:
+          path: C:\Users\runneradmin\.conan2
+          key: windows-2019-conan-${{ hashFiles('**/conanfile.py') }}
+          restore-keys: |
+            windows-2019-conan-
+      - name: Setup and build
+        run: |
+          make setup
+          make module.tar.gz
+      # As of April 2025, there isn't an obvious way to upload to the Viam module registry from
+      # Windows. So, we upload to the Github Actions artifacts, then download it on Linux in a later
+      # step to upload to the module registry.
+      - name: Store the module as a temporary artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: windows-2019-module-${{ github.sha }}
+          path: ./module.tar.gz
 
-      - uses: actions/cache@v3
+  build-unix:
+    name: Build for Unix
```

Comment:
I know macOS is unix, but could you make the name more clear that it's both Linux and macOS? "Build for Linux and Darwin" (or macOS). At glance it looks just like it's a build for Linux.

Reply:
Heh, you looked at this too quickly! I noticed that the Windows and Unix versions were very similar except for the cache directory, so have refactored into a giant build step for everything, and updated the name.